src/mlpack/tests/main_tests/krann_test.cpp

```diff
@@ -0,0 +1,479 @@
+  output_model = std::move(CLI::GetParam<RANNModel*>("output_model"));
+
+  // Reset passed parameters.
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+
+  // Input saved model, pass the same query and keep k unchanged.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 5);
+  SetInputParam("leaf_size", (int) 10);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Check that initial output matrices and the output matrices using
+  // saved model are equal.
+  BOOST_CHECK_EQUAL(output_model->LeafSize(), (int) 1);
+  BOOST_CHECK_EQUAL(CLI::GetParam<RANNModel*>("output_model")->LeafSize(),
+      (int) 10);
+  delete output_model;
+}
+
+/**
+ * Ensure that different tau give different results.
+ */
+BOOST_AUTO_TEST_CASE(KRANNDifferentTau)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Random input, some k <= number of top tau %ile reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("k", (int) 5);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+  
+  RANNModel* output_model;
+  output_model = std::move(CLI::GetParam<RANNModel*>("output_model"));
+  
+  // Reset the passed  parameters.
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+
+  // Changing value of tau and keeping everything else unchanged.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 5);
+  SetInputParam("tau", (double) 10);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Check that initial output matrices and the output matrices using
+  // saved model are equal
+  BOOST_CHECK_EQUAL(output_model->Tau(), (double) 5);
+  BOOST_CHECK_EQUAL(CLI::GetParam<RANNModel*>("output_model")->Tau(),
+        (double) 10);
+  delete output_model;      
+}
+
+/**
+ * Ensure that different alpha give different results.
+ */
+BOOST_AUTO_TEST_CASE(KRANNDifferentAlpha)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Random input, some k <= number of top tau %ile reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("k", (int) 5);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+  
+  RANNModel* output_model;
+  output_model = std::move(CLI::GetParam<RANNModel*>("output_model"));
+  
+  // Reset the passed  parameters.
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+
+  // Changing value of tau and keeping everything else unchanged.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 5);
+  SetInputParam("alpha", (double) 0.80);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Check that initial output matrices and the output matrices using
+  // saved model are equal
+  BOOST_CHECK_EQUAL(output_model->Alpha(), (double) 0.95);
+  BOOST_CHECK_EQUAL(CLI::GetParam<RANNModel*>("output_model")->Alpha(),
+        (double) 0.80);
```

Comment:
```suggestion
      (double) 0.80);
```
In our style guide a wrapped line only needs to be indented twice. :)
https://github.com/mlpack/mlpack/wiki/DesignGuidelines#style-guidelines